### ISSUE TITLE
Add a reference for some of the most common environment variables

### ISSUE THREAD

**Issue by sharifsalah** [AcademySoftwareFoundation/opencue.io#131]
Add a reference for some of the most common environment variables, such as:

CUESUBMIT_CONFIG_FILE
OL_CONFIG
CUE_FRAME_LOG_DIR
CUEBOT_HOSTS

These are just some examples as there are many others, but it would be good to publish something basic and add to it over time.

Please add any suggestions for useful variables to this issue.

**Comment by bcipriano** [comment]
We should also document the various tokens that can be used in the commands you send to Cue.

We can pull the list from the spot in the code where those tokens are replaced: https://github.com/AcademySoftwareFoundation/OpenCue/blob/master/cuebot/src/main/java/com/imageworks/spcue/dispatcher/DispatchSupportService.java#L390

https://github.com/AcademySoftwareFoundation/OpenCue/pull/597 just added a few new ones.

**Comment by apoorvkhare07** [comment]
@bcipriano @sharifsalah Can I work on this issue if it's still relevant and open? 
If so, can you guide me on how to work on this?

**Comment by apoorvkhare07** [comment]
@sharifsalah I would like to work on this issue if it's available ?

**Comment by sharifsalah** [comment]
@apoorvkhare07 sure, thanks for volunteering! I'll assign you the issue.

**Comment by apoorvkhare07** [comment]
@sharifsalah Thank you. As per my understanding I need to add the environment related variables to docker.yaml for all except CueBOT and to Dockerfile for CueBOT. right ? 
Also should I provide them with default values for now ?

**Comment by sharifsalah** [comment]
@apoorvkhare07 I'm not sure quite what you mean. Docker is optional for all components of OpenCue. Users might use Docker to install the project but others might not use it. It will depend on the setup.

So this issue isn't strictly related to managing OpenCue in a container environment. We want to document a reference of what all the environment variables and tokens do.

For example, to document the tokens, we might use a list similar to the following:

## Tokens

(explain what a token is)

You can use the following tokens when submitting outline jobs, either in
the graphical Cuesubmit client, or via the PyOutline API:

* `#ZFRAME#`: (I can't remember what this token does right now)
* `#IFRAME#`: The number of the current frame.
* `#FRAME_START#`: The number of the final frame in the current chunk.
* `#FRAME_END#`: The number of the final frame in the current chunk.
* `#FRAME_CHUNK#`: The value of the current chunk.
* `#LAYER#"`: The name of the current layer.
* `#JOB#"`: The name of the current job.
* `#FRAMESPEC#`: The value of the current frame spec. A frame spec consists

**Comment by apoorvkhare07** [comment]
Ok, Got the point. 
So where in the documentation do I need to add these references ?

**Comment by apoorvkhare07** [comment]
https://www.opencue.io/docs/reference/ here we don't a page for CueBOT and tokens as well.

**Comment by sharifsalah** [comment]
Yes, I think adding a new page under https://www.opencue.io/docs/reference/  would be good.

To clarify, when you say a page for CueBot and tokens: this issue is for environment variables for all of OpenCue, not just CueBot and tokens. See the original description for more details. Eventually this page will include a reference for all the variables in the description as well as others we might not have identified yet. 

Don't feel you have to document all of these. For example, a basic page that describes the tokens (as I explained in my previous comment), would be very welcome and useful. We can create additional pull requests later on.

We strongly recommend you send smaller pull requests as this could be quite a large page eventually.
